<launch>
  <arg name="simulation" default="false"/>

  <include unless="$(arg simulation)" file="$(find thorp_manipulation)/launch/includes/controllers.launch.xml">
    <!-- here we only load physical robot controllers, so no needed on simulation -->
  </include>

  <node pkg="arbotix_controllers" type="gripper_controller.py" name="gripper_controller" output="screen" respawn="true">
    <!--
     Gripper controller parameters:
     more than the physical properties of the gripper, these values are an ad-hoc configuration
     that works with arbotix_ros/gripper_controller/OneSideGripperModel
      -->
    <param name="model" value="singlesided"/>
    <param name="invert" value="true"/>
    <param name="center" value="0.07"/>
    <param name="pad_width" value="0.025"/>
    <param name="finger_length" value="0.03"/>
    <param name="min_opening" value="0.0025"/>  <!-- more closed than this provokes gazebo_rospkgs issue #479! -->
    <param name="max_opening" value="0.05"/>
    <param name="stalled_time" value="0.5"/>
  </node>

  <!-- Fake gripper link joint state publisher -->
  <node pkg="turtlebot_arm_bringup" type="fake_joint_pub.py" name="fake_gripper_joint_pub" output="screen" respawn="true"/>
</launch>
